mrmustard/utils/wigner.py

```diff
@@ -18,53 +18,158 @@
 from numba import njit
 
 from mrmustard import settings
+from mrmustard.math import Math
 
 
-def wigner_discretized(rho, qvec, pvec):
+__all__ = ["wigner_discretized"]
+
+math = Math()
+
+# ~~~~~~~
+# Helpers
+# ~~~~~~~
+
+
+@njit
+def make_grid(q_vec, p_vec, hbar):  # pragma: no cover
+    r"""Returns two coordinate matrices `Q` and `P` from coordinate vectors
+    `q_vec` and `p_vec`, along with the grid over which Wigner functions can be
+    discretized.
+    """
+    Q = np.outer(q_vec, np.ones_like(p_vec))
+    P = np.outer(np.ones_like(q_vec), p_vec)
+    return Q, P, (Q + P * 1.0j) / np.sqrt(2 * hbar)
+
+
+@njit
+def _wig_laguerre_val(L, x, diag):  # pragma: no cover
+    """Returns the coefficient `c_L = sum_n rho_{n,L+n} Z_n^L` used
+    by `_wigner_discretized_clenshaw`. The evaluation uses the Clenshaw recursion.
+    """
+    if len(diag) == 2:
+        y0 = np.array([[diag[0]]]).astype(np.complex128)
```

Comment:
instead of changing type after the fact you can use `np.array([[diag[0]]], dtype=np.complex128)`